Firmware/Marlin_main.cpp

```diff
@@ -3817,14 +3800,13 @@ static void gcode_M600(bool automatic, float x_position, float y_position, float
             current_position[E_AXIS] += FILAMENTCHANGE_RECFEED;
             plan_buffer_line_curposXYZE(FILAMENTCHANGE_EXFEED);
         }
-    
+
+        // Recover Z axis
```

Comment:
we shall first return XY and then move Z down - it's been that way before for a reason

Reply:
Moving the Z after XY wasn’t working in my testing with `raise_z` which is why I moved the Z before XY, then it suddenly worked.

I think it has something to do with the XY move using `current_position[Z_AXIS]` because using raise_z after the XY move, no movement happens.

Reply:
:thinking: this doesn't sound right, there might be some other problem hidden in the back.
At least let's add a comment about this issue, because we'll run into it again later.

Reply:
Ok I’ll add a TODO in a few minutes :)

Reply:
Done